server/client.go

```diff
@@ -146,6 +147,15 @@ func (c *client) GetOpts() *clientOpts {
 	return &c.opts
 }
 
+func (c *client) GetTLSConnectionState() *tls.ConnectionState {
+	tc, ok := c.nc.(*tls.Conn)
+	if !ok {
+		return nil
+	}
+	state := tc.ConnectionState()
```

Comment:
The underlying function (https://golang.org/pkg/crypto/tls/#Conn.ConnectionState) already uses a lock during state preparation.

I think the ConnectionState, in the crypto/tls api, is meant for read-only operations, and modifying the slices in it may lead to serious problems, but nothing is done to avoid that.

Question is: should we protect something crypto/tls does not bother to protect?